src/shared/selinux-util.c

```diff
@@ -413,8 +413,18 @@ int mac_selinux_get_our_label(char **label) {
 
 int mac_selinux_get_child_mls_label(int socket_fd, const char *exe, const char *exec_label, char **label) {
 #if HAVE_SELINUX
-        _cleanup_freecon_ char *mycon = NULL, *peercon = NULL, *fcon = NULL;
-        _cleanup_context_free_ context_t pcon = NULL, bcon = NULL;
+        /* mycon — SELinux context under which systemd is running
+         * (e.g.: systemd_u:systemd_r:init_t:s0-s3.c0-c1023);
+         * peercon — SELinux context of the socket,
+         * from which we get the MLS level (the last part of the context)
+         * (e.g.: systemd_u:object_r:netlabel_peer_t:s0);
+         * targetcon — computed target context that should be set without correct MLS level;
+         * resultcon — computed target context that should be set with correct MLS level;
+         * tcon — internal target SELinux context;
+         * pcon — internal peercon SELinux context;
+         * bcon — internal SELinux context set as mycon, but with MLS level of peercon. */
```

Comment:
Need for such a comment means we are doing something wrong. What about having 3 functions like,

```C
static int mac_selinux_get_selinux_context_mls_label(const char *mls_range, char **ret);
static int mac_selinux_get_exe_mls_label(const char *mls_range, char **ret);
static int mac_selinux_get_manager_mls_label(const char *mls_range, char **ret);
```
Then we would get peer context and using that we would only call appropriate function (based on situation we are in) and return its result. This would simplify this function and would eliminated the need for this context variables.